hummingbot/client/config/config_data_types.py

```diff
@@ -0,0 +1,47 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Optional, Callable
+
+from pydantic import BaseModel
+from pydantic.schema import default_ref_template
+
+from hummingbot.client.config.config_helpers import strategy_config_schema_encoder
+
+
+class ClientConfigEnum(Enum):
+    def __str__(self):
+        return self.value
+
+
+@dataclass()
+class ClientFieldData:
+    prompt: Optional[Callable[['BaseClientModel'], str]] = None
+    prompt_on_new: bool = False
+
+
+class BaseClientModel(BaseModel):
+    """
+    Notes on configs:
+    - In nested models, be weary that pydantic will take the first model that fits
+      (see https://pydantic-docs.helpmanual.io/usage/model_config/#smart-union).
+    """
+    @classmethod
+    def schema_json(
+        cls, *, by_alias: bool = True, ref_template: str = default_ref_template, **dumps_kwargs: Any
+    ) -> str:
+        # todo: make it ignore `client_data` all together
```

Comment:
We should check if the todo is still valid

Reply:
Yes, it's still valid, but I did not want to spend more time on it now as the solution is not immediately obvious, the functionality will mostly be used for hs (eventually), and it can still be used as is by ignoring the extra client-specific metadata that will be added to the schema